### PR TITLE
[Projects] Add Iguazio version validation for adding node selector to project

### PR DESCRIPTION
**Pull request by yaelgen**
Adds a check for the Iguazio version before applying the default_function_node_selector to project attributes to ensures backwards compatibility.
 - Introduce getter/setter methods for default_function_node_selector.
 - Add check for the Iguazio version when transforming mlrun project to iguazio project

resolves: [ML-6316](https://iguazio.atlassian.net/browse/ML-6316)

[ML-6316]: https://iguazio.atlassian.net/browse/ML-6316?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: mlrun/utils/helpers.py [hunk 1]
```diff
@@ -983,6 +983,12 @@ def get_workflow_url(project, id=None):
     return url
 
 
+def is_igz_version_sufficient(required_version):
```
Comment (alonmr) [inline mlrun/utils/helpers.py]: no need for a new method you can use `mlrun.platforms.iguazio.min_iguazio_versions` or `mlrun.utils.helpers.validate_component_version_compatibility`

File: tests/api/utils/clients/test_iguazio.py [hunk 2]
```diff
@@ -937,6 +943,7 @@ async def test_format_as_leader_project(
             exclude_paths=[
                 "root['attributes']['updated_at']",
                 "root['attributes']['operational_status']",
+                # "root['attributes']['default_function_node_selector']"
```
Comment (alonmr) [inline tests/api/utils/clients/test_iguazio.py]: remove comment?

File: server/api/utils/clients/iguazio.py [hunk 3]
```diff
@@ -752,11 +752,22 @@ def _transform_mlrun_project_to_iguazio_project(
...
+                        project.spec.default_function_node_selector
+                    )
                 )
-            )
+            else:
+                logger.warning(
+                    "User attempted to use a feature requiring project-level default_function_node_selector,"
+                    " but the installed version of Iguazio is not compatible. "
+                    "Project created without the field in Iguazio."
+                )
```
Comment (alonmr) [inline server/api/utils/clients/iguazio.py]: Warning is a bit much for my taste, suggest giving less worrying log and log some informative parameters
```suggestion
                logger.debug(
                    "Omitting project-level default function node selector from Iguazio project as Iguazio version is insufficient", 
                    igz_version=mlconf.igz_version,
                    project_name= project.metadata.name
                )
```

**Review by alonmr — CHANGES_REQUESTED** [review]
Nice! Some small comments

File: mlrun/utils/helpers.py (hunk 1 above)
Comment (yaelgen) [reply mlrun/utils/helpers.py]: nice thx 👍

File: server/api/utils/clients/iguazio.py [hunk 4]
```diff
@@ -755,17 +755,18 @@ def _transform_mlrun_project_to_iguazio_project(
             # This feature requires support for project-level default_function_node_selector,
             # which is available starting from version 3.5.5.
             # We are adding this validation to maintain backward compatibility with older versions of Iguazio.
-            if mlrun.utils.helpers.is_igz_version_sufficient("3.5.5"):
+            if mlrun.utils.helpers.validate_component_version_compatibility("3.5.5"):
```
Comment (alonmr) [inline server/api/utils/clients/iguazio.py]: I think you missed the 1st param
```suggestion
            if mlrun.utils.helpers.validate_component_version_compatibility("iguazio", "3.5.5"):
```

**Review by alonmr — APPROVED** [review]

**Pull request merged**